Avoid to call AC_TRY_RUN

Upstream-Status: Inappropriate [configuration]

Avoid to call AC_TRY_RUN to check if GSSAPI libraries support SPNEGO
on cross-compile environment by definition AC_ARG_ENABLE enable-spnego

Signed-off-by: Roy.Li <rongqing.li@windriver.com>
---
 cmulocal/sasl2.m4 |   13 ++++++++++++-
 1 file changed, 12 insertions(+), 1 deletion(-)

diff --git a/cmulocal/sasl2.m4 b/cmulocal/sasl2.m4
index 3c2841a..a5ecf81 100644
--- a/cmulocal/sasl2.m4
+++ b/cmulocal/sasl2.m4
@@ -281,6 +281,17 @@ if test "$gssapi" != no; then
 
   cmu_save_LIBS="$LIBS"
   LIBS="$LIBS $GSSAPIBASE_LIBS"
+  AC_ARG_ENABLE([spnego],
+              [AC_HELP_STRING([--enable-spnego=<DIR>],
+                              [enable SPNEGO support in GSSAPI libraries [no]])],
+              [spnego=$enableval],
+              [spnego=no])
+
+  if test "$spnego" = no; then
+       echo "no"
+  elif test "$spnego" = yes; then
+       AC_DEFINE(HAVE_GSS_SPNEGO,,[Define if your GSSAPI implementation supports SPNEGO])
+  else
   AC_MSG_CHECKING([for SPNEGO support in GSSAPI libraries])
   AC_TRY_RUN([
 #ifdef HAVE_GSSAPI_H
@@ -308,7 +319,7 @@ int main(void)
 	AC_MSG_RESULT(yes) ],
 	AC_MSG_RESULT(no))
   LIBS="$cmu_save_LIBS"
-
+   fi
 else
   AC_MSG_RESULT([disabled])
 fi
-- 
1.7.10.4

